File: spring-boot-oauth2-auth-server/src/main/java/com/bittokazi/oauth2/auth/server/app/services/role/helpers/RoleHelpers.java
package com.bittokazi.oauth2.auth.server.app.services.role.helpers;

import com.bittokazi.oauth2.auth.server.app.models.tenant.Role;
import com.bittokazi.oauth2.auth.server.app.repositories.tenant.RoleRepository;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class RoleHelpers {

    public static Object getRoles(int page, int count, RoleRepository roleRepository) {
        Map<String, Object> json = new HashMap<>();
        Pageable reqCount = PageRequest.of(page, count, Sort.by(Sort.Direction.DESC, "id"));
        Page<Role> pages = roleRepository.findAll(reqCount);
        json.put("roles", pages.getContent());
        json.put("pages", pages.getTotalPages());
        json.put("records", pages.getTotalElements());
        return json;
    }

    public static Object getAllRoles(RoleRepository roleRepository) {
        Map<String, Object> json = new HashMap<>();
        List<Role> roles = roleRepository.findAll();
        json.put("roles", roles);
        json.put("pages", 1);
        json.put("records", roles.size());
        return json;
    }
}

File: spring-boot-oauth2-auth-server/src/main/kotlin/com/bittokazi/oauth2/auth/server/app/services/role/helpers/RoleHelpers.kt
package com.bittokazi.oauth2.auth.server.app.services.role.helpers

import com.bittokazi.oauth2.auth.server.app.models.tenant.RoleList
import com.bittokazi.oauth2.auth.server.app.repositories.tenant.RoleRepository
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Sort

object RoleHelpers {
    fun getRoles(page: Int, count: Int, roleRepository: RoleRepository): Any {
        val reqCount: Pageable = PageRequest.of(page, count, Sort.by(Sort.Direction.DESC, "id"))
        val pages = roleRepository.findAll(reqCount)
        return RoleList(
            pages = pages.totalPages,
            records = pages.totalElements,
            roles = pages.content
        )
    }

    fun getAllRoles(roleRepository: RoleRepository): Any {
        val roles = roleRepository.findAll()
        return RoleList(
            pages = 1,
            records = roles.size.toLong(),
            roles = roles
        )
    }
}
